dbms/src/Common/SensitiveDataMasker.cpp

```diff
@@ -0,0 +1,166 @@
+#include "SensitiveDataMasker.h"
+
+#include <set>
+#include <string>
+#include <atomic>
+
+#include <re2/re2.h>
+#include <re2/stringpiece.h>
+
+#include <Poco/Util/AbstractConfiguration.h>
+
+#include <common/logger_useful.h>
+
+#include <Common/Exception.h>
+#include <Common/StringUtils/StringUtils.h>
+
+#ifndef NDEBUG
+#    include <iostream>
+#endif
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int CANNOT_COMPILE_REGEXP;
+    extern const int NO_ELEMENTS_IN_CONFIG;
+    extern const int INVALID_CONFIG_PARAMETER;
+}
+
+class SensitiveDataMasker::MaskingRule
+{
+private:
+    const std::string name;
+    const std::string replacement_string;
+    const std::string regexp_string;
+
+    const RE2 regexp;
+    const re2::StringPiece replacement;
+
+    mutable std::atomic<std::uint64_t> matches_count = 0;
```

Comment:
Can we remove this? QueryMaskingRulesMatch should be sufficient to check whether the rules are working.

Reply:
Is was used in unit tests, can hide that in #ifdef